src/redux/books/Books.js

```diff
@@ -0,0 +1,43 @@
+const addBook = 'BOOK_ADDED';
+const removeBook = 'BOOK_REMOVED';
+
+export const addBookFunc = (payload) => ({
+  type: addBook,
+  payload,
+});
+
+export const removeBookFunc = (id) => ({
+  type: removeBook,
+  id,
+});
+
+// Reducer
+
+const bookReducer = (state = [], action) => {
+  switch (action.type) {
+    case addBook:
+      return [...state, { ...action.payload }];
+
+    case removeBook:
+      return state.filter((book) => book.id !== action.id);
+
```

Comment:
- [ ] Here, the id attribute which you want to compare with `book.id` can not be accessed through `action.id` but rather through `action.payload.id`. I suggest you do it like this 👇 
```
case removeBook:
      return state.filter((book) => book.id !== action.payload.id);
```